tests/test_agent_output_schemas.py

```diff
@@ -0,0 +1,630 @@
+                                    'type': 'object',
+                                },
+                            },
+                            'required': ['kind', 'data'],
+                            'additionalProperties': False,
+                            'title': 'int',
+                        },
+                    ]
+                }
+            },
+            'required': ['result'],
+            'additionalProperties': False,
+        }
+    )
+
+
+async def test_tool_output_json_schema():
+    agent = Agent(
+        'test',
+        output_type=[ToolOutput(bool)],
+    )
+    assert agent.output_json_schema() == snapshot(
+        {'type': 'object', 'properties': {'response': {'type': 'boolean'}}, 'required': ['response']}
+    )
+
+    agent = Agent(
+        'test',
+        output_type=[ToolOutput(bool, name='alice', description='Dreaming...')],
+    )
+    assert agent.output_json_schema() == snapshot(
+        {'type': 'object', 'properties': {'response': {'type': 'boolean'}}, 'required': ['response']}
+    )
+
+    agent = Agent(
+        'test',
+        output_type=[ToolOutput(bool), ToolOutput(Foo)],
+    )
+    assert agent.output_json_schema() == snapshot(
+        {
+            'type': 'object',
+            'properties': {
+                'result': {
+                    'anyOf': [
+                        {
+                            'type': 'object',
+                            'properties': {
+                                'kind': {'type': 'string', 'const': 'final_result_bool'},
+                                'data': {
+                                    'properties': {'response': {'type': 'boolean'}},
+                                    'required': ['response'],
+                                    'type': 'object',
+                                },
+                            },
+                            'required': ['kind', 'data'],
+                            'additionalProperties': False,
+                        },
+                        {
+                            'type': 'object',
+                            'properties': {
+                                'kind': {'type': 'string', 'const': 'final_result_Foo'},
+                                'data': {
+                                    'properties': {
+                                        'a': {'items': {'$ref': '#/$defs/Bar'}, 'type': 'array'},
+                                        'b': {'type': 'integer'},
+                                    },
+                                    'required': ['a', 'b'],
+                                    'type': 'object',
+                                },
+                            },
+                            'required': ['kind', 'data'],
+                            'additionalProperties': False,
+                            'title': 'Foo',
+                        },
+                    ]
+                }
+            },
+            'required': ['result'],
```

Comment:
Same notes as above, we should end up with a schema representing `bool | Foo` without all the wrapper keys